### PR TITLE
[DOC-604] Download always latest oasisctl tag in Dockerfile

### PR DESCRIPTION
**Pull request by dandimeo**
#### Upstream PRs

- 3.10:
- 3.11:
- 3.12:

### CODE REVIEW

**Review by Simran-B — CHANGES_REQUESTED** [review]
This still requires that we rebuild (and publish?) the image. The oasisctl generator should instead fetch the latest release and download the files so that we always get the latest oasisctl in the scheduled workflow without rebuilding the image.

**Comment by dandimeo** [comment]
> This still requires that we rebuild (and publish?) the image. The oasisctl generator should instead fetch the latest release and download the files so that we always get the latest oasisctl in the scheduled workflow without rebuilding the image.

Oasisctl latest binary download is now part of the toolchain script so it will always run when "oasisctl" is part of generators.

https://app.circleci.com/pipelines/github/arangodb/docs-hugo/2737/workflows/7a78e746-966f-4320-9547-ac0b0ad7b324/jobs/6968

Quite basic implementation for the moment. A future better implementation could do some version checking to avoid duplicating the download step for each docs version

File: toolchain/scripts/toolchain.sh [hunk 1]
```diff
@@ -551,6 +553,18 @@ function generate_oasisctl() {
   log "[generate_oasisctl] Done"
 }
 
+function download_oasisctl() {
+  oasisctlVersion=$(curl -I https://github.com/arangodb-managed/oasisctl/releases/latest | awk -F '/' '/^location/ {print  substr($NF, 1, length($NF)-1)}')
+  log "[download_oasisctl] Downloading oasisctl version $oasisctlVersion"
+  cd /tmp
+  rm -r oasisctl.zip bin
```
Comment (Simran-B) [inline toolchain/scripts/toolchain.sh:560]: When the generation runs for 3.10, 3.11, and 3.12, then for the first version (3.10) the files do not exist. In the 3.11 and 3.12 runs, they do exist. Based on this observation, can't we skip the download of the same oasisctl version for 3.11 and 3.12 simply by checking if we already have the binary?

Comment (dandimeo) [reply toolchain/scripts/toolchain.sh]: Solved in Commit https://github.com/arangodb/docs-hugo/pull/288/commits/319810ac42f8472774abe4146492143c6f6afd3e

oasisctl is downloaded only if /tmp/oasisctl.zip is not present already and once for the entire toolchain run
https://app.circleci.com/pipelines/github/arangodb/docs-hugo/2750/workflows/148bb9f5-9ef7-41ee-a606-3f43a2a6cd82/jobs/7010

File: toolchain/scripts/toolchain.sh [hunk 2]
```diff
@@ -521,7 +521,11 @@ function generate_oasisctl() {
   version=$1
 
   log "[generate_oasisctl] Generate OasisCTL docs"
+  log "[generate_oasisctl] Invoking download_oasisctl"
```
Comment (Simran-B) [inline toolchain/scripts/toolchain.sh:524]: This should be moved inside the if

**Review by Simran-B — APPROVED** [review]
LGTM, only logging should be slightly adjusted

File: toolchain/scripts/toolchain.sh (hunk 2 above)
Comment (dandimeo) [reply toolchain/scripts/toolchain.sh]: Solved in https://github.com/arangodb/docs-hugo/pull/288/commits/841bec51be755533ad43ed2491e89bbb680d0e9d

**Pull request merged**